auditor/perturbations/paraphrase.py

```diff
@@ -0,0 +1,97 @@
+from typing import List, Optional
+import os
+import re
+
+import openai
+
+from auditor.perturbations.base import AbstractPerturbation
+from auditor.perturbations.constants import OPENAI_CHAT_COMPLETION
+
+
+class Paraphrase(AbstractPerturbation):
+    """Perturbation class that paraphrases by querying open-ai LLM
+    """
+    def __init__(
+        self,
+        model: Optional[str] = OPENAI_CHAT_COMPLETION,
+        num_sentences: int = 5,
+        temperature: float = 0.0,
+        api_key: Optional[str] = None,
+        api_version: Optional[str] = None,
+    ) -> None:
+        self._init_key(api_key)
+        self._init_model(model, api_version)
+        self.num_sentences = num_sentences
+        self.temperature = temperature
+        self.descriptor = (
+            f'Paraphrases the original prompt with '
+            f'an open-ai {self.model} model.'
+        )
+        self.paraphrase_instruction = (
+            'Generate a bulleted list of {n} sentences '
+            'with same meaning as \"{sentence}\"'
+        )
+        return
+
+    def description(self) -> str:
+        return self.descriptor
+
+    def _init_key(self, api_key: str):
+        """Initialize API key"""
+        if api_key is None:
+            api_key = os.getenv("OPENAI_API_KEY")
+        self.api_key = api_key
+        openai.api_key = api_key
+        return
+
+    def _init_model(
+        self,
+        model,
+        api_version
+    ):
+        """Initialize model, engine and api version"""
+        self.model = model
+        self.api_version = api_version
+        if openai.api_type == "azure":
+            self.engine = model
+            self.api_version = api_version
+        else:
+            self.engine = None
+        return
+
+    def perturb(
+        self,
+        prompt: str,
+    ) -> List[str]:
+        prompt = self.paraphrase_instruction.format(
+            n=self.num_sentences,
+            sentence=prompt
+        )
+        payload = [
+            {
+                "role": "user",
+                "content": prompt
+            }
+        ]
+        response = openai.ChatCompletion.create(
+            model=self.model,
+            messages=payload,
+            temperature=self.temperature,
+            engine=self.engine,
+            api_version=self.api_version,
+        )
+        return Paraphrase._process_similar_sentence_reponse(response)
+
+    @staticmethod
+    def _process_similar_sentence_reponse(response):
```

Comment:
@mshergad This should address the bug you reported for paraphrasing single sentences. At a later point we should add a class that can paraphrase multi-line prompts.